pandas/tslib.pyx

```diff
@@ -2292,6 +2292,13 @@ cpdef array_to_datetime(ndarray[object] values, errors='raise',
                         iresult[i] = cast_from_unit(val, 'ns')
                     except:
                         iresult[i] = NPY_NAT
+            elif is_bool_object(val) and not is_coerce:
+                if val == NPY_NAT:
+                    iresult[i] = NPY_NAT
+                elif is_raise:
+                    raise TypeError("boolean is not convertible to datetime")
+                else:
+                    return values
```

Comment:
```val == NPY_NAT``` is copied from just above (for int and float processing)

Your simplified version works :), thank you. I just pushed up the new commit